### ISSUE TITLE
Channel spam(?) on my node spiking write IOPS to 35 bringing down my cloud node.

### ISSUE THREAD

**Issue by jarret** [ElementsProject/lightning#2762]
This successfully brought down my AWS-hosted node the other day which was running `bitcoind` and my lightning-dir on a (admittedly quite inexpensive) magnetic "cold" drive (EBS `sc1`-class 500GB drive on AWS). It spiked write IOPS to ~35 and consumed all my burst credits and stalled everything to do with that drive since the IO was getting heavily throttled in the datacenter from there.

I moved the lightning-dir to a SSD drive that was better able to keep up and it started again this morning and stopped a little while ago and I was able to capture it happening:

The debug output looks like this, scrolling by very quickly in a loop:

```
2019-06-20T14:25:59.886Z lightning_hsmd(32026): Client: Received message 19 from client
2019-06-20T14:25:59.886Z lightningd(32017): lightning_channeld-024b9a1fa8e006f1e3937f65f66c408e6da8e1ca728ea43222a7381df1cc449605 chan #4458: Creating commit_sig signature 87307 304402205295068a9cba8a313aff897285c536b4c22089f256f590bcfc804f59d87ee6ae02202b30713537e686b2eb14205784b3e2f852bd1916a34d77fe9045abc6ed69571001 for tx 0200000001c864f37fd1e39261488d782b0e0b60d6bc0b65925f8c96153fe6526f8c7675ed0000000000a727d78002eb09000000000000160014fbb5087d1689c2db17a40d80a18e8bbaedcfbf1a48c2020000000000220020cf40d085578d52de27e1f6b7b42f7db2e3c1070843f7026dd1916dfadca64d50196f7620 wscript 52210212cec95e663f232c2bcf9f255fb300352ecff55b8f1df8e8dc0e625dffcc57c321027e23e944d2e6fcda0b5b716e36df636e9562f5194fae4292c4dd5b8871ce837352ae key 0212cec95e663f232c2bcf9f255fb300352ecff55b8f1df8e8dc0e625dffcc57c3
2019-06-20T14:25:59.889Z lightningd(32017): lightning_channeld-024b9a1fa8e006f1e3937f65f66c408e6da8e1ca728ea43222a7381df1cc449605 chan #4458: Log pruned 881 entries (mem 131536 -> 16526)
2019-06-20T14:25:59.889Z lightningd(32017): lightning_channeld-024b9a1fa8e006f1e3937f65f66c408e6da8e1ca728ea43222a7381df1cc449605 chan #4458: Telling master we're about to commit...
2019-06-20T14:25:59.889Z lightningd(32017): lightning_channeld-024b9a1fa8e006f1e3937f65f66c408e6da8e1ca728ea43222a7381df1cc449605 chan #4458: Sending master 1020
2019-06-20T14:25:59.889Z lightningd(32017): lightning_channeld-024b9a1fa8e006f1e3937f65f66c408e6da8e1ca728ea43222a7381df1cc449605 chan #4458: UPDATE WIRE_CHANNEL_SENDING_COMMITSIG
2019-06-20T14:25:59.895Z lightningd(32017): lightning_channeld-024b9a1fa8e006f1e3937f65f66c408e6da8e1ca728ea43222a7381df1cc449605 chan #4458: ... , awaiting 1120
2019-06-20T14:25:59.895Z lightningd(32017): lightning_channeld-024b9a1fa8e006f1e3937f65f66c408e6da8e1ca728ea43222a7381df1cc449605 chan #4458: Got it!
2019-06-20T14:25:59.895Z lightningd(32017): lightning_channeld-024b9a1fa8e006f1e3937f65f66c408e6da8e1ca728ea43222a7381df1cc449605 chan #4458: Sending commit_sig with 0 htlc sigs
2019-06-20T14:25:59.895Z lightningd(32017): lightning_channeld-024b9a1fa8e006f1e3937f65f66c408e6da8e1ca728ea43222a7381df1cc449605 chan #4458: peer_out WIRE_COMMITMENT_SIGNED
2019-06-20T14:26:00.326Z lightningd(32017): lightning_channeld-024b9a1fa8e006f1e3937f65f66c408e6da8e1ca728ea43222a7381df1cc449605 chan #4458: peer_in WIRE_REVOKE_AND_ACK
2019-06-20T14:26:00.326Z lightningd(32017): lightning_channeld-024b9a1fa8e006f1e3937f65f66c408e6da8e1ca728ea43222a7381df1cc449605 chan #4458: Received revoke_and_ack
2019-06-20T14:26:00.326Z lightningd(32017): lightning_channeld-024b9a1fa8e006f1e3937f65f66c408e6da8e1ca728ea43222a7381df1cc449605 chan #4458: No commits outstanding after recv revoke_and_ack
2019-06-20T14:26:00.326Z lightningd(32017): lightning_channeld-024b9a1fa8e006f1e3937f65f66c408e6da8e1ca728ea43222a7381df1cc449605 chan #4458: Sending master 1022
2019-06-20T14:26:00.326Z lightningd(32017): lightning_channeld-024b9a1fa8e006f1e3937f65f66c408e6da8e1ca728ea43222a7381df1cc449605 chan #4458: UPDATE WIRE_CHANNEL_GOT_REVOKE
2019-06-20T14:26:00.326Z lightningd(32017): 024b9a1fa8e006f1e3937f65f66c408e6da8e1ca728ea43222a7381df1cc449605 chan #4458: got revoke 87306: 0 changed
2019-06-20T14:26:00.330Z lightningd(32017): lightning_channeld-024b9a1fa8e006f1e3937f65f66c408e6da8e1ca728ea43222a7381df1cc449605 chan #4458: ... , awaiting 1122
2019-06-20T14:26:00.331Z lightningd(32017): lightning_channeld-024b9a1fa8e006f1e3937f65f66c408e6da8e1ca728ea43222a7381df1cc449605 chan #4458: Got it!
2019-06-20T14:26:00.331Z lightningd(32017): lightning_channeld-024b9a1fa8e006f1e3937f65f66c408e6da8e1ca728ea43222a7381df1cc449605 chan #4458: revoke_and_ack LOCAL: remote_per_commit = 03a920acdff71d93b13de55c8dedc53850d7ab88c64d045cf7efd40da0217a3485, old_remote_per_commit = 03f37c9df7533f5c044c7987cc34ee8d1a7617c313805af2aa1dbf8cb43263cb32
2019-06-20T14:26:00.331Z lightningd(32017): lightning_channeld-024b9a1fa8e006f1e3937f65f66c408e6da8e1ca728ea43222a7381df1cc449605 chan #4458: Setting REMOTE feerate to 23000
2019-06-20T14:26:00.331Z lightningd(32017): lightning_channeld-024b9a1fa8e006f1e3937f65f66c408e6da8e1ca728ea43222a7381df1cc449605 chan #4458: peer_out WIRE_UPDATE_FEE
2019-06-20T14:26:00.331Z lightningd(32017): lightning_channeld-024b9a1fa8e006f1e3937f65f66c408e6da8e1ca728ea43222a7381df1cc449605 chan #4458: Trying commit
2019-06-20T14:26:00.333Z lightning_hsmd(32026): Client: Received message 19 from client
2019-06-20T14:26:00.333Z lightningd(32017): lightning_channeld-024b9a1fa8e006f1e3937f65f66c408e6da8e1ca728ea43222a7381df1cc449605 chan #4458: Creating commit_sig signature 87308 3044022044f165f433dacaa6fe2fa66df46a12273eb573c9d2b8c81d3c7e626918ac1fee022062a8fb0958053062e63af06365c6f7c80da5440da1f4f0fcdb57cfb14dc37b8201 for tx 0200000001c864f37fd1e39261488d782b0e0b60d6bc0b65925f8c96153fe6526f8c7675ed0000000000a727d78002eb09000000000000160014644d47fc8b9e51f0c6fefa29d0e963f3fb65510a48c202000000000022002073ccfe1a80a98296d205c7949715bfc3fec4e3137e6c669a2194769a6879d1c01e6f7620 wscript 52210212cec95e663f232c2bcf9f255fb300352ecff55b8f1df8e8dc0e625dffcc57c321027e23e944d2e6fcda0b5b716e36df636e9562f5194fae4292c4dd5b8871ce837352ae key 0212cec95e663f232c2bcf9f255fb300352ecff55b8f1df8e8dc0e625dffcc57c3
2019-06-20T14:26:00.333Z lightningd(32017): lightning_channeld-024b9a1fa8e006f1e3937f65f66c408e6da8e1ca728ea43222a7381df1cc449605 chan #4458: Telling master we're about to commit...
2019-06-20T14:26:00.333Z lightningd(32017): lightning_channeld-024b9a1fa8e006f1e3937f65f66c408e6da8e1ca728ea43222a7381df1cc449605 chan #4458: Sending master 1020
2019-06-20T14:26:00.333Z lightningd(32017): lightning_channeld-024b9a1fa8e006f1e3937f65f66c408e6da8e1ca728ea43222a7381df1cc449605 chan #4458: UPDATE WIRE_CHANNEL_SENDING_COMMITSIG
2019-06-20T14:26:00.337Z lightningd(32017): lightning_channeld-024b9a1fa8e006f1e3937f65f66c408e6da8e1ca728ea43222a7381df1cc449605 chan #4458: ... , awaiting 1120
2019-06-20T14:26:00.337Z lightningd(32017): lightning_channeld-024b9a1fa8e006f1e3937f65f66c408e6da8e1ca728ea43222a7381df1cc449605 chan #4458: Got it!
2019-06-20T14:26:00.337Z lightningd(32017): lightning_channeld-024b9a1fa8e006f1e3937f65f66c408e6da8e1ca728ea43222a7381df1cc449605 chan #4458: Sending commit_sig with 0 htlc sigs
2019-06-20T14:26:00.337Z lightningd(32017): lightning_channeld-024b9a1fa8e006f1e3937f65f66c408e6da8e1ca728ea43222a7381df1cc449605 chan #4458: peer_out WIRE_COMMITMENT_SIGNED
2019-06-20T14:26:00.767Z lightningd(32017): lightning_channeld-024b9a1fa8e006f1e3937f65f66c408e6da8e1ca728ea43222a7381df1cc449605 chan #4458: peer_in WIRE_REVOKE_AND_ACK
2019-06-20T14:26:00.767Z lightningd(32017): lightning_channeld-024b9a1fa8e006f1e3937f65f66c408e6da8e1ca728ea43222a7381df1cc449605 chan #4458: Received revoke_and_ack
2019-06-20T14:26:00.767Z lightningd(32017): lightning_channeld-024b9a1fa8e006f1e3937f65f66c408e6da8e1ca728ea43222a7381df1cc449605 chan #4458: No commits outstanding after recv revoke_and_ack
2019-06-20T14:26:00.767Z lightningd(32017): lightning_channeld-024b9a1fa8e006f1e3937f65f66c408e6da8e1ca728ea43222a7381df1cc449605 chan #4458: Sending master 1022
2019-06-20T14:26:00.767Z lightningd(32017): lightning_channeld-024b9a1fa8e006f1e3937f65f66c408e6da8e1ca728ea43222a7381df1cc449605 chan #4458: UPDATE WIRE_CHANNEL_GOT_REVOKE
2019-06-20T14:26:00.767Z lightningd(32017): 024b9a1fa8e006f1e3937f65f66c408e6da8e1ca728ea43222a7381df1cc449605 chan #4458: got revoke 87307: 0 changed
2019-06-20T14:26:00.771Z lightningd(32017): lightning_channeld-024b9a1fa8e006f1e3937f65f66c408e6da8e1ca728ea43222a7381df1cc449605 chan #4458: ... , awaiting 1122
2019-06-20T14:26:00.772Z lightningd(32017): lightning_channeld-024b9a1fa8e006f1e3937f65f66c408e6da8e1ca728ea43222a7381df1cc449605 chan #4458: Got it!
2019-06-20T14:26:00.772Z lightningd(32017): lightning_channeld-024b9a1fa8e006f1e3937f65f66c408e6da8e1ca728ea43222a7381df1cc449605 chan #4458: revoke_and_ack LOCAL: remote_per_commit = 020dd16147262f0bab5f0072d17e53625895ad1f453fd886d759c0eddce2fb31d1, old_remote_per_commit = 03a920acdff71d93b13de55c8dedc53850d7ab88c64d045cf7efd40da0217a3485
2019-06-20T14:26:00.772Z lightningd(32017): lightning_channeld-024b9a1fa8e006f1e3937f65f66c408e6da8e1ca728ea43222a7381df1cc449605 chan #4458: Setting REMOTE feerate to 23000
2019-06-20T14:26:00.772Z lightningd(32017): lightning_channeld-024b9a1fa8e006f1e3937f65f66c408e6da8e1ca728ea43222a7381df1cc449605 chan #4458: peer_out WIRE_UPDATE_FEE
2019-06-20T14:26:00.772Z lightningd(32017): lightning_channeld-024b9a1fa8e006f1e3937f65f66c408e6da8e1ca728ea43222a7381df1cc449605 chan #4458: Trying commit
2019-06-20T14:26:00.774Z lightning_hsmd(32026): Client: Received message 19 from client
```
The scroll seems to be referencing nodeID `024b9a1fa8e006f1e3937f65f66c408e6da8e1ca728ea43222a7381df1cc449605`

which is listed as:

```
{
  "nodes": [
    {
      "nodeid": "024b9a1fa8e006f1e3937f65f66c408e6da8e1ca728ea43222a7381df1cc449605", 
      "alias": "BLUEIRON-.1rc2-6-g5e3690b-modded", 
      "color": "024b9a", 
      "last_timestamp": 1560831305, 
      "globalfeatures": "", 
      "global_features": "", 
      "addresses": [
        {
          "type": "ipv4", 
          "address": "128.199.202.168", 
          "port": 9735
        }, 
        {
          "type": "ipv6", 
          "address": "2400:6180:0:d0::5cd2:a001", 
          "port": 9735
        }, 
        {
          "type": "torv2", 
          "address": "pehpzf3erua23pa7.onion", 
          "port": 9735
        }, 
        {
          "type": "torv3", 
          "address": "ki5uack5xb4gicfnpfavzdibojfu3qvwtlg3ca7eyjxik7tgwfxummqd.onion", 
          "port": 9735
        }
      ]
    }
  ]
}
```

I do have my application also heartbeating to that drive with some very minor file IO, but the spike is purely coming from the lightning-dir, as that is the only other thing on the drive. Here is what AWS's monitor looks like for that period for reads and writes:

writes IOPS:
![write-iops](https://user-images.githubusercontent.com/528853/59890266-6410b880-938d-11e9-9b97-675fbfdda43d.png)

read IOPS:
![read-iops](https://user-images.githubusercontent.com/528853/59890285-73900180-938d-11e9-8a34-9920de25fcfa.png)

### `getinfo` output

```
{
  "id": "02e389d861acd9d6f5700c99c6c33dd4460d6f1e2f6ba89d1f4f36be85fc60f8d7", 
  "alias": "Sparkshot.io Station Zebra", 
  "color": "ffc45a", 
  "num_peers": 30, 
  "num_pending_channels": 5, 
  "num_active_channels": 25, 
  "num_inactive_channels": 0, 
  "address": [
    {
      "type": "ipv4", 
      "address": "54.188.180.102", 
      "port": 9735
    }
  ], 
  "binding": [
    {
      "type": "ipv4", 
      "address": "172.31.38.43", 
      "port": 9735
    }
  ], 
  "version": "v0.7.0", 
  "blockheight": 581665, 
  "network": "bitcoin", 
  "msatoshi_fees_collected": 7524, 
  "fees_collected_msat": "7524msat"
}
```

**Comment by jarret** [comment]
Forgot to include the fact, I do have a channel open with node `024b9a1fa8e006f1e3937f65f66c408e6da8e1ca728ea43222a7381df1cc449605`:
```{
  "peers": [
    {
      "id": "024b9a1fa8e006f1e3937f65f66c408e6da8e1ca728ea43222a7381df1cc449605", 
      "connected": true, 
      "netaddr": [
        "128.199.202.168:9735"
      ], 
      "global_features": "", 
      "local_features": "aa", 
      "globalfeatures": "", 
      "localfeatures": "aa", 
      "channels": [
        {
          "state": "CHANNELD_NORMAL", 
          "scratch_txid": "076fade18454348abc14ded702af1276c9057913ff0264ec135c5434aa36cae8", 
          "owner": "lightning_channeld", 
          "short_channel_id": "564104x1696x0", 
          "direction": 1, 
          "channel_id": "c864f37fd1e39261488d782b0e0b60d6bc0b65925f8c96153fe6526f8c7675ed", 
          "funding_txid": "ed75768c6f52e63f15968c5f92650bbcd6600b0e2b788d486192e3d17ff364c8", 
          "private": false, 
          "funding_allocation_msat": {
            "024b9a1fa8e006f1e3937f65f66c408e6da8e1ca728ea43222a7381df1cc449605": 0, 
            "02e389d861acd9d6f5700c99c6c33dd4460d6f1e2f6ba89d1f4f36be85fc60f8d7": 200000000
          }, 
          "funding_msat": {
            "024b9a1fa8e006f1e3937f65f66c408e6da8e1ca728ea43222a7381df1cc449605": "0msat", 
            "02e389d861acd9d6f5700c99c6c33dd4460d6f1e2f6ba89d1f4f36be85fc60f8d7": "200000000msat"
          }, 
          "msatoshi_to_us": 19191976, 
          "to_us_msat": "19191976msat", 
          "msatoshi_to_us_min": 19191976, 
          "min_to_us_msat": "19191976msat", 
          "msatoshi_to_us_max": 200000000, 
          "max_to_us_msat": "200000000msat", 
          "msatoshi_total": 200000000, 
          "total_msat": "200000000msat", 
          "dust_limit_satoshis": 546, 
          "dust_limit_msat": "546000msat", 
          "max_htlc_value_in_flight_msat": 18446744073709551615, 
          "max_total_htlc_in_msat": "18446744073709551615msat", 
          "their_channel_reserve_satoshis": 2000, 
          "their_reserve_msat": "2000000msat", 
          "our_channel_reserve_satoshis": 2000, 
          "our_reserve_msat": "2000000msat", 
          "spendable_msatoshi": 17191976, 
          "spendable_msat": "17191976msat", 
          "htlc_minimum_msat": 0, 
          "minimum_htlc_in_msat": "0msat", 
          "their_to_self_delay": 144, 
          "our_to_self_delay": 144, 
          "max_accepted_htlcs": 483, 
          "status": [
            "CHANNELD_NORMAL:Reconnected, and reestablished.", 
            "CHANNELD_NORMAL:Funding transaction locked. Channel announced."
          ], 
          "in_payments_offered": 5, 
          "in_msatoshi_offered": 1992479, 
          "in_offered_msat": "1992479msat", 
          "in_payments_fulfilled": 0, 
          "in_msatoshi_fulfilled": 0, 
          "in_fulfilled_msat": "0msat", 
          "out_payments_offered": 25, 
          "out_msatoshi_offered": 2346449785, 
          "out_offered_msat": "2346449785msat", 
          "out_payments_fulfilled": 6, 
          "out_msatoshi_fulfilled": 180808024, 
          "out_fulfilled_msat": "180808024msat", 
          "htlcs": [
          ]
        }
      ]
    }
  ]
}
```

**Comment by ZmnSCPxj** [comment]
We are doing lots of disc writes because we are receiving lots of `revoke_and_ack`, which strongly requires that we write to disk very quickly. We are getting lots of `revoke_and_ack` because we send a lot of `update_fee`s.

 Can you provide a more complete log? Even just 1 minute or so, starting just before the spike? We should not be updating fees in milliseconds.

**Comment by jarret** [comment]
I don't have the full log from this occurrence, but I just set it up to log to a file. Will be monitoring closely and post the log if/when it happens again.

**Comment by jarret** [comment]
Didn't take long for it to start again.

![write-iops2](https://user-images.githubusercontent.com/528853/59893218-75f85880-9399-11e9-9ff9-73ec93212c94.png)

Here are the logs from before the spike and for a couple minutes during https://gist.github.com/jarret/611b9b42afd31f4d6af366bdb0b4d405

Seems to start at:  `2019-06-21T01:49:36.919Z`

**Comment by rustyrussell** [comment]
Heh, this is my node :)  And you're spamming me, not the other way around.

Fortunately, this is a known issue, fixed in #2718

**Comment by jarret** [comment]
Sorry, I'll keep 'em on a tighter leash when around other nodes. ;)

Thanks @rustyrussell @ZmnSCPxj
